src/watts/plugin_moose.py

```diff
@@ -205,3 +211,24 @@ def postrun(self, params: Parameters) -> ResultsMOOSE:
         inputs = ['MOOSE.i']
         outputs = [p for p in Path.cwd().iterdir() if p.name not in inputs]
         return ResultsMOOSE(params, time, inputs, outputs)
+
+
+    def _convert_unit(self, params: Parameters):
+        # Function to convert units to SI
+        u.imperial.enable()
+        self.params_copy = copy.deepcopy(params)
+
+        for key in self.params_copy.keys():
+            if key in self.params_copy.convert_param:
```

Comment:
Rather than keeping track of `convert_param`, you can just have a check on the type of the value:
```Python
if isinstance(self.params_copy[key], astropy.units.Quantity):
    ...
```